pkg/testexecutionservice/testexecution.go

```diff
@@ -118,9 +118,10 @@ func (tes *testExecutionService) Run(ctx context.Context,
 		tes.logger.Errorf("failed to find process for command %s with pid %d %v", cmd.String(), pid, err)
 		return nil, err
 	}
-	if err := cmd.Wait(); err != nil {
-		tes.logger.Errorf("Error in executing []: %+v\n", err)
-		return nil, err
+
+	errC := cmd.Wait()
+	if errC != nil {
+		tes.logger.Errorf("Error in executing []: %+v\n", errC)
 	}
 	executionResults := <-tes.ts.ExecutionResultOutputChannel
```

Comment:
couldn't this be a blocking call, in case no  goroutine one writes to this channel `ExecutionResultOutputChannel`. As a precaution we can add select {case: default:}

Reply:
we want it to be blocking here. The default case handling is already there in `teststats.go` which is sending messages to this channel